model/histogram/generic.go

```diff
@@ -28,33 +28,41 @@ const (
 	CustomBucketsSchema          int32 = -53
 )
 
+type Error struct {
```

Comment:
There is a test called `TestHistogramValidation` . Let's modify it so that when it expects an error it also checks that error.As  is true for Error.